cadquery/cq.py

```diff
@@ -371,6 +374,10 @@ def _computeXdir(normal):
                     raise ValueError(
                         "Needs a face or a vertex or point on a work plane")
 
+        # update center to projected origin if desired
+        if centerOption == 'ProjectedOrigin':
+            center = Vector(0, 0, 0).projectToPlane(center, normal)
```

Comment:
I am a little bit confused - I thought that you intended to project current origin (i.e. `self.plane.origin`) onto the new plane. Here it seems that you are projecting a hardcoded point `Vector(0,0,0)` instead. Is this indeed the intention?

Reply:
The intent is to project the global origin onto the workplane. The use case where I needed this is when I'm using an unusually shaped face to create a workplane (see picture below where I'm using the highlighted faces to create a new workplane). The center of mass or bounding box center of the selected faces are not intuitive to use as an origin for additional features. However, if I can project the origin (highlighted in red in the figure below) onto the workplane defined by the faces, I have a convenient center to reference for creating the pockets. The projection code is general and can project an arbitrary point to the surface. However, I thought a common use case would be to project the global origin onto the workplane to define its center. Often when I'm creating parts in SolidWorks, I'll reference dimensions off of the global origin in a similar fashion.

![image](https://user-images.githubusercontent.com/6439649/58731477-6b5b2c80-83b4-11e9-948d-e8b59a19abac.png)

Reply:
OK, got it. If you use `self.plane.origin` it will be more generic. The default value is `(0,0,0)` so it will support your use case but it will also allow users to to project other origin values specified with a `moveTo(x,y).workplane()` call.

Reply:
I don't think this will work since the moveTo(x,y) commands will accumulate if we do this and the absolute origin will be lost without reversing all of the previous moveTo(x,y) calls in the stack. Changing Vector(0,0,0) to self.plane.origin results in the part shown below (the user space code is unmodified only the suggested change to the workplane() method from (0,0,0) to self.plane.origin was made). The first pocket comes out fine because it is after the first move. The other pockets end up off of the part since the move commands are accumulating and the absolute origin is lost.

![image](https://user-images.githubusercontent.com/6439649/58736471-9d27bf80-83c3-11e9-8a19-33f39dc5f331.png)

If I understand your suggestion correctly, you can accomplish what you're suggesting by leaving the projection from Vector(0,0,0) in the workplane() method and using the moveTo after creating the workplane: `workplane(centerOption='ProjectedOrigin').moveTo(x,y)` This is the approach I'm using in the code to create the pockets in the part.

Reply:
Weird, could you share the code used to generate your part? I'd like to investigate

Reply:
In addition to `moveTo()` calls, I had used `transformed()` calls with the offset option, which changes `plane.origin` (I'm still learning how all this works). I did this because I wanted to rotate the workplane for certain features. You're right, if I had used only `moveTo()` calls, my code would probably have worked both ways (I did a simple test to confirm this). 

Would it be better to call my proposed option `workplane(centerOption='ProjectedGlobalOrigin')` rather than `workplane(centerOption='ProjectedOrigin')` and then have the `workplane(centerOption='ProjectedOrigin')` use `self.plane.origin` like you suggest? I want the ability to go back to the global origin even if I've used transformed() or center() calls.

Reply:
I updated the pull request with the change to have both the `ProjectedOrigin` option that uses `self.plane.origin` and the `ProjectedGlobalOrigin` that uses `Vector(0,0,0)`. I think part of the confusion was me using the term origin when I really meant global origin. The behavior of these two options is only different if there has been a call to `transformed` (with the offset option) or to `center`, both of which update the current plane's origin (there may be other methods that update the plane origin that I'm not aware of). The examples below illustrate the difference.

I've also updated to pull request to incorporate the resolved review comments.

**Using moveTo with ProjectedOrigin**
```
import cadquery as cq

model = cq.Workplane('XY').rect(100,100).extrude(10)
model = model.faces('>Z').workplane(centerOption='ProjectedOrigin').moveTo(-40,-40).hole(5)
model = model.faces('>Z').workplane(centerOption='ProjectedOrigin').moveTo(40,40).hole(5)
```
![image](https://user-images.githubusercontent.com/6439649/58754944-c0c03800-849f-11e9-9302-f592b5205ce0.png)

**Using moveTo with ProjectedGlobalOrigin (no difference)**
```
import cadquery as cq

model = cq.Workplane('XY').rect(100,100).extrude(10)
model = model.faces('>Z').workplane(centerOption='ProjectedGlobalOrigin').moveTo(-40,-40).hole(5)
model = model.faces('>Z').workplane(centerOption='ProjectedGlobalOrigin').moveTo(40,40).hole(5)
```
![image](https://user-images.githubusercontent.com/6439649/58754974-3f1cda00-84a0-11e9-91eb-489e4d758a89.png)

**Using center with ProjectedOrigin (changes result)**
```
import cadquery as cq

model = cq.Workplane('XY').rect(100,100).extrude(10)
model = model.faces('>Z').workplane(centerOption='ProjectedOrigin').center(-40,-40).hole(5)
model = model.faces('>Z').workplane(centerOption='ProjectedOrigin').center(40,40).hole(5)
```
![image](https://user-images.githubusercontent.com/6439649/58754991-65db1080-84a0-11e9-9e5d-64b5ec60e800.png)

**Using center with ProjectedGlobalOrigin (same result as when using moveTo)**
```
import cadquery as cq

model = cq.Workplane('XY').rect(100,100).extrude(10)
model = model.faces('>Z').workplane(centerOption='ProjectedGlobalOrigin').center(-40,-40).hole(5)
model = model.faces('>Z').workplane(centerOption='ProjectedGlobalOrigin').center(40,40).hole(5)
```
![image](https://user-images.githubusercontent.com/6439649/58754999-9d49bd00-84a0-11e9-8dc4-4af40011a6af.png)